python/pyspark/taskcontext.py

```diff
@@ -95,3 +96,33 @@ def getLocalProperty(self, key):
         Get a local property set upstream in the driver, or None if it is missing.
         """
         return self._localProperties.get(key, None)
+
+    def barrier(self):
```

Comment:
Create `BarrierTaskContext` that extends `TaskContext` and then move those two methods there.